syz-ci/manager_test.go

```diff
@@ -106,3 +119,129 @@ Reported-by: foo+abcd000@bar.com`,
 	assert.Equal(t, commit.Title, "title with fix")
 	assert.ElementsMatch(t, commit.BugIDs, []string{"abcd000"})
 }
+
+func TestUploadCoverJSONLToGCS(t *testing.T) {
```

Comment:
Nit (just came to my mind)

FWIW one other option to test the function is to mock not the `gcs.Client` object, but rather `gcs.UploadFile`. Your code in `manager.go` is relying only on that function, so you are actually mocking one level deeper than could be enough.

`UploadFile` could probably be turned into some type to simplify mocking (`UploadService`?).

Reply:
Agree, there are alternatives here. I didn't consider it.
Thinking about the services isolation looks more natural to me.

Deeper coverage - better? :)

Reply:
I think ideally we should better have the tests for the `syz-ci` code in the `syz-ci` package and the tests for the `pkg/gcs` code in the `pkg/gcs` package.

Reply:
> We currently misuse cloud.google.com/go/storage in pkg/gcs

How exactly?

> Use pkg/gcs singleton.

Why singleton?

Reply:
I quoted the documentation. Client should be reused. We don't reuse it.

Reply:
Ah, right. The quote looked like a quote from some discussion and that confused me.

Yes, it sounds reasonable to have just one client. Not sure whether we really need a singleton for that (that would only complicate testing). You could e.g. create the client in `main()` and pass it down to each `Manager` object.

https://github.com/google/syzkaller/blob/20510e8871e31b24ba7dfd19ee03a482ae2ad722/syz-ci/syz-ci.go#L287

These changes though now seem to be totally separate from what this PR was created for :)